tests/unit/s2n_mutual_auth_test.c

```diff
@@ -328,8 +328,88 @@ int main(int argc, char **argv)
+        struct s2n_stuffer server_to_client;
+
+        /* Craft a cipher preference with a cipher_idx cipher
+           NOTE: Its safe to use memcpy as the address of server_cipher_preferences
+           will never be NULL */
+        memcpy(&server_cipher_preferences, default_cipher_preferences, sizeof(server_cipher_preferences));
+        server_cipher_preferences.count = 1;
+        struct s2n_cipher_suite *cur_cipher = default_cipher_preferences->suites[cipher_idx];
+
+        if (!cur_cipher->available) {
+            /* Skip Ciphers that aren't supported with the linked libcrypto */
+            continue;
+        }
+
+        server_cipher_preferences.suites = &cur_cipher;
+        config->cipher_preferences = &server_cipher_preferences;
+
+        EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_SERVER));
+        EXPECT_SUCCESS(s2n_connection_set_config(server_conn, config));
+
+        EXPECT_NOT_NULL(client_conn = s2n_connection_new(S2N_CLIENT));
+        EXPECT_SUCCESS(s2n_connection_set_config(client_conn, config));
+
+        /* Only set S2N_CERT_AUTH_REQUIRED on the server and not the client so that the connection fails */
+        EXPECT_SUCCESS(s2n_connection_set_client_auth_type(server_conn, S2N_CERT_AUTH_REQUIRED));
+
+
+        /* Set up our I/O callbacks. Use stuffers for the "I/O context" */
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&client_to_server, 0));
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&server_to_client, 0));
+
+        /* Set Up Callbacks*/
+        EXPECT_SUCCESS(s2n_connection_set_recv_cb(client_conn, &buffer_read));
+        EXPECT_SUCCESS(s2n_connection_set_send_cb(client_conn, &buffer_write));
+        EXPECT_SUCCESS(s2n_connection_set_recv_cb(server_conn, &buffer_read));
+        EXPECT_SUCCESS(s2n_connection_set_send_cb(server_conn, &buffer_write));
+
+        /* Set up Callback Contexts to use stuffers */
+        EXPECT_SUCCESS(s2n_connection_set_recv_ctx(client_conn, &server_to_client));
+        EXPECT_SUCCESS(s2n_connection_set_send_ctx(client_conn, &client_to_server));
+        EXPECT_SUCCESS(s2n_connection_set_recv_ctx(server_conn, &client_to_server));
+        EXPECT_SUCCESS(s2n_connection_set_send_ctx(server_conn, &server_to_client));
+
+        int tries = 0;
+        int failures = 0;
+        do {
+            int client_ret, server_ret;
+            client_ret = s2n_negotiate(client_conn, &client_blocked);
+            server_ret = s2n_negotiate(server_conn, &server_blocked);
+            tries += 1;
+
+            if (client_ret != 0 || server_ret != 0) {
+               failures ++;
+            }
+        } while ((client_blocked || server_blocked) && tries < MAX_TRIES);
+
+        EXPECT_EQUAL(failures, MAX_TRIES);
+        /* Verify that NEITHER connections negotiated Mutual Auth */
+        EXPECT_FALSE(s2n_connection_is_client_authenticated(server_conn));
+        EXPECT_FALSE(s2n_connection_is_client_authenticated(client_conn));
```

Comment:
Okay, awesome. That's what I was hoping. I'd be happy with that name change then.